qiskit/qpy/binary_io/value.py

```diff
@@ -475,8 +472,9 @@ def _read_parameter_expression(file_obj):
         *struct.unpack(formats.PARAMETER_EXPR_PACK, file_obj.read(formats.PARAMETER_EXPR_SIZE))
     )
 
-    sympy_str = file_obj.read(data.expr_size).decode(common.ENCODE)
-    expr_ = symengine.sympify(parse_sympy_repr(sympy_str))
+    from sympy.parsing.sympy_parser import parse_expr
+
+    expr_ = parse_expr(file_obj.read(data.expr_size).decode(common.ENCODE))
```

Comment:
This is definitely not something we can do, it is undoing #14024 and reintroducing https://github.com/Qiskit/qiskit/security/advisories/GHSA-6m2c-76ff-6vrf we can't use sympy's `parse_expr()` anywhere arbitrary input is accepted.

Reply:
Fixed by [af8a219](https://github.com/Qiskit/qiskit/pull/13278/commits/af8a219e9abaaaa4b3cacd8e3e53d6b9dcae0cce)